timer: cortex_m_systick: Compilation error if CONFIG_TIMER_READS_ITS_FREQUENCY_AT_RUNTIME is defined

### Describe the bug

Timer cortex_m_systick fails to build if CONFIG_TIMER_READS_ITS_FREQUENCY_AT_RUNTIME is defined

Regression of commit b9a39a77d37e105043177c3b732daf7db661d5fb

### Regression

- [x] This is a regression.

### Steps to reproduce

west build -p -b qemu_cortex_m3 samples/hello_world -- -DCONFIG_TIMER_READS_ITS_FREQUENCY_AT_RUNTIME=y

### Relevant log output

```shell
/home/user/projects/zephyrproject/zephyr/drivers/timer/cortex_m_systick.c:23:48: error: division by zero in #if
   23 | #define CYC_PER_TICK (z_clock_hw_cycles_per_sec/CONFIG_SYS_CLOCK_TICKS_PER_SEC)
      |                                                ^
/home/user/projects/zephyrproject/zephyr/drivers/timer/cortex_m_systick.c:33:20: note: in expansion of macro 'CYC_PER_TICK'
   33 | #if (COUNTER_MAX / CYC_PER_TICK) == 1
```

### Impact

Not sure

### Environment

v4.30 Release

### Additional Context

_No response_

--- a/drivers/timer/cortex_m_systick.c
+++ b/drivers/timer/cortex_m_systick.c
@@ -20,19 +20,19 @@
 
 #if defined(CONFIG_TIMER_READS_ITS_FREQUENCY_AT_RUNTIME)
 extern unsigned int z_clock_hw_cycles_per_sec;
+/* CYC_PER_TICK must be inside of systick capacities (<1Ghz) */
 #define CYC_PER_TICK (z_clock_hw_cycles_per_sec/CONFIG_SYS_CLOCK_TICKS_PER_SEC)
 #else
 #define CYC_PER_TICK (CONFIG_SYS_CLOCK_HW_CYCLES_PER_SEC/CONFIG_SYS_CLOCK_TICKS_PER_SEC)
+#if (COUNTER_MAX / CYC_PER_TICK) == 1
+#pragma message("tickless does nothing as CONFIG_SYS_CLOCK_TICKS_PER_SEC too low")
+#endif
 #endif
 
 /* add MAX_TICKS protection */
 #define _MAX_TICKS (int)((k_ticks_t)(COUNTER_MAX / CYC_PER_TICK) - 1)
 #define MAX_TICKS ((_MAX_TICKS > 0) ? _MAX_TICKS : 1)
 #define MAX_CYCLES (MAX_TICKS * CYC_PER_TICK)
-
-#if (COUNTER_MAX / CYC_PER_TICK) == 1
-#pragma message("tickless does nothing as CONFIG_SYS_CLOCK_TICKS_PER_SEC too low")
-#endif
 
 /* Minimum cycles in the future to try to program.  Note that this is
  * NOT simply "enough cycles to get the counter read and reprogrammed

